core/mempool/txblock_builder.go

```diff
@@ -0,0 +1,87 @@
+package mempool
+
+import (
+	"idena-go/blockchain/types"
+	"idena-go/common"
+)
+
+type buildingContext struct {
+	sortedTxs          []*types.Transaction
+	sortedPriorityTxs  []*types.Transaction
+	sortedTxsPerSender map[common.Address][]*types.Transaction
+	curNoncesPerSender map[common.Address]uint32
+	blockTxs           []*types.Transaction
+	blockSize          int
+}
+
+func newBuildingContext(sortedTxs []*types.Transaction,
+	sortedPriorityTxs []*types.Transaction,
+	sortedTxsPerSender map[common.Address][]*types.Transaction,
+	curNoncesPerSender map[common.Address]uint32) *buildingContext {
+
+	ctx := &buildingContext{
+		sortedTxs:          sortedTxs,
+		sortedPriorityTxs:  sortedPriorityTxs,
+		sortedTxsPerSender: sortedTxsPerSender,
+		curNoncesPerSender: curNoncesPerSender,
+	}
+	return ctx
+}
+
+func (ctx *buildingContext) addPriorityTxsToBlock() {
+	for len(ctx.sortedPriorityTxs) > 0 {
```

Comment:
Maybe we need to stop when block max size is reached, because now we will iterate over all txs even if max block size is reached